### PR TITLE
Modify rule S3649(PHP): Update text to the Education Framework structure (APPSEC-190)

### PR DESCRIPTION
**Pull request by loris-s-sonarsource**

### CODE REVIEW

File: docs/allowed_framework_names.adoc [hunk 1]
```diff
@@ -20,6 +20,7 @@
 * Core PHP
 * Laravel
 * Symfony
+* CakePHP
```
Comment (loris-s-sonarsource) [inline docs/allowed_framework_names.adoc:23]:
```suggestion
```

File: rules/S3649/common/fix/secure-by-design.adoc [hunk 2]
```diff
@@ -0,0 +1,15 @@
+==== Use secure APIs
+
+Some libraries provide a database framework that frees the developer from
+sanitizing or writing prepared statements.
```
Comment (pedro-oliveira-sonarsource) [inline rules/S3649/common/fix/secure-by-design.adoc:4]:
```suggestion
Some frameworks provide a database abstraction layer that frees the developers
 from sanitizing or writing prepared statements.
```

File: rules/S3649/common/fix/secure-by-design.adoc [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+These APIS can be described as "secure by design". +
```
Comment (pedro-oliveira-sonarsource) [inline rules/S3649/common/fix/secure-by-design.adoc:6]:
```suggestion
These provided APIs can be described as "secure by design". +
```

File: rules/S3649/common/fix/secure-by-design.adoc [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+By providing a builder pattern with PDO parameter binding behind the scenes,
```
Comment (pedro-oliveira-sonarsource) [inline rules/S3649/common/fix/secure-by-design.adoc:7]:
```suggestion
By providing a builder pattern with parameter binding behind the scenes,
```

Comment (pedro-oliveira-sonarsource) [reply rules/S3649/common/fix/secure-by-design.adoc:7]: To make it generic as PDO is specific to PHP to be reusable on other languages.

File: rules/S3649/common/fix/secure-by-design.adoc [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+`{secure_feature}` can be called "secure by design" as it adds
+multiple layers of security to the code while keeping the codebase shorter.
+
+*Note*: This type of API can also provide "raw" functions or equivalents. These
```
Comment (pedro-oliveira-sonarsource) [inline rules/S3649/common/fix/secure-by-design.adoc:11]:
```suggestion
*Note*: These type of APIs can also provide "raw" functions or equivalents. These
```

Comment (pedro-oliveira-sonarsource) [reply rules/S3649/common/fix/secure-by-design.adoc:11]: To be consistent with the plural on the previous paragraph.

File: rules/S3649/php/how-to-fix-it/laravel.adoc [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        $users = DB::select($query); // Noncompliant
+
+        if (count($users) != 1)
+        {
+            abort(401);
+        }
+
+        return view('authenticated.index');
+    }
+}
```
[the remark this answers is not in the document]
Comment (pedro-oliveira-sonarsource) [reply rules/S3649/php/how-to-fix-it/laravel.adoc:32]: As already discussed, the code samples are all good, not sure whether they should be simplified.

File: rules/S3649/common/fix/secure-by-design.adoc (hunk 5 above)
Comment (loris-s-sonarsource) [reply rules/S3649/common/fix/secure-by-design.adoc:11]:
```suggestion
*Note*: These types of APIs can also provide "raw" functions or equivalents. These
```

File: rules/S3649/php/how-to-fix-it/laravel.adoc (hunk 6 above)
[the remark this answers is not in the document]
Comment (loris-s-sonarsource) [reply rules/S3649/php/how-to-fix-it/laravel.adoc:32]: let's keep the one already written and discuss the direction we want to take during a retro or a circke

**Review by pedro-oliveira-sonarsource — APPROVED** [review]
LGTM.

File: rules/S3649/common/fix/secure-by-design.adoc [hunk 7]
```diff
@@ -0,0 +1,15 @@
+==== Use secure APIs
+
+Some frameworks provide a database abstraction layer that frees the developers
+ from sanitizing or writing prepared statements.
```
Comment (loris-s-sonarsource) [inline rules/S3649/common/fix/secure-by-design.adoc:4]:
```suggestion
from sanitizing or writing prepared statements.
```

**Pull request merged**